external/webserver/http_client.c

```diff
@@ -838,10 +907,30 @@ int http_send_response_helper(struct http_client_t *client, int status, const ch
 		if (headers) {
 			cur = headers->head->next;
 			while (cur != headers->tail) {
+				if (strcmp(cur->key, "Content-Length") == 0 || strcmp(cur->key, "Content-Type") == 0
+					|| strcmp(cur->key, "Keep-Alive") == 0) {
+					cur = cur->next;
+					continue;
+				}
+
 				buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen,
 								   "%s: %s\r\n", cur->key, cur->value);
 				cur = cur->next;
 			}
+
+			// Add content type and content length headers
+			if (body) {
+				buflen += snprintf(buf + buflen,
```

Comment:
snprintf can be failed. then this code caused unexpected behavior. So how about to check validation of snprintf then update buflen.